bpf/bpf_lxc.c

```diff
@@ -778,7 +789,8 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx,
 #endif /* ENABLE_ROUTING */
 			policy_clear_mark(ctx);
 			return ipv4_local_delivery(ctx, l3_off, SECLABEL, ip4,
-						   ep, METRIC_EGRESS, false);
+						   ep, METRIC_EGRESS,
+						   bypass_ingress_policy);
```

Comment:
This does not seem correct, we're changing the `from_host` argument instead of `hairpin_flow`, which was added in fcbd5d780bc5085f4cc24f197abaab0be2989143.

I'm unsure whether we should also backport this other commit or not. My understanding is that it should address a similar issue to what we're trying to fix with the current backport, but when per-endpoint routes are _not_ enabled. I suppose that without a compelling reason, we should avoid pulling more commits for this branch, so I'd just drop this hunk (and the previous one where we initialise `bypass_ingress_policy`).

Cc @aditighag FYI.

Reply:
removed that hunk and this change. @qmonnet look good to you now?